tests/manual-test-cases/Group5-Functional-Tests/5-22-NFS-Volume.md

```diff
@@ -0,0 +1,83 @@
+Test 5-22 - NFS Volume
+=======
+
+# Purpose:
+To verify that NFS shared volumes work with currently supported docker commands
+
+# References:
+[1 - Best practices for running VMware vSphere on NFS](http://www.vmware.com/content/dam/digitalmarketing/vmware/en/pdf/techpaper/vmware-nfs-bestpractices-white-paper-en.pdf)
+
+[2 - Docker Command Line Reference - Volume Create](https://docs.docker.com/engine/reference/commandline/volume_create/)
+
+[3 - Docker Command Line Reference - Exec](https://docs.docker.com/engine/reference/commandline/exec/)
+
+[4 - Docker Command Line Reference - Volume Inspect](https://docs.docker.com/engine/reference/commandline/volume_inspect/)
+
+[5 - Docker Command Line Reference](https://docs.docker.com/engine/reference/commandline/volume_ls/)
+
+
+# Environment:
+This test requires access to VMware Nimbus cluster for dynamic ESXi, vCenter, and NFS server creation
+
+# Test Steps:
+1. Deploy VIC appliance to vCenter with a simple cluster and one NFS server; nfs share point added here.
```

Comment:
I will chime in here as well: 

1. Correct, no readonly shares, it might be possible in some scenarios where the volume store was created and configured the exact way a customer wanted. Then we would only join existing containers(probably a conversation for another place/time).

2. Correct, we only support NFSv3. On deployment we would not fail the `vic-machine create` command since we cannot know that it is not mountable until portlayer init time. What we would expect is a warning message from the vic-machine output stating that volume stores x,y, and z were not established successfully. It will also point the customer/user to the portlayer logs for more explanation. We should make sure this is the case. This will encompass the test for trying to establish a store that is NFSv4. Vic-machine should warn you that the store was not created, and you should find a line in the portlayer logs stating `Invalid Protocol` during the volume store init process. 

3. I think it is important to see how we behave when the nfs server falls over after containers are running, though we do not manage the runtime after the container starts. So for that scenario we would expect weird and bad things to start happening to the container process and potentially the r/w layer as well since it is possible logging/writing for that volume could fallback to the underlying FS depending on the running app. The other problem with this, and this is has a ticket open for it, is that the tether will fail silently on NFS mounts that fail at the moment, which can lead customers/users to think they are writing to the nfs server when they are not.